static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -11,18 +11,54 @@ import { isBlocked } from 'common/modules/commercial/membership-engagement-banne
 import { getSync as getGeoLocation } from 'lib/geolocation';
 import { shouldShowReaderRevenue } from 'common/modules/commercial/contributions-utilities';
 import type { Banner } from 'common/modules/ui/bannerPicker';
+import bean from 'bean';
+import fetchJson from 'lib/fetch-json';
+import reportError from 'lib/report-error';
 
 import {
     submitComponentEvent,
     addTrackingCodesToUrl,
 } from 'common/modules/commercial/acquisitions-ophan';
 import { acquisitionsBannerControlTemplate } from 'common/modules/commercial/templates/acquisitions-banner-control';
+import userPrefs from 'common/modules/user-prefs';
 
-// change messageCode to force redisplay of the message to users who already closed it.
-const messageCode = 'engagement-banner-2018-07-13';
+type BannerDeployLog = {
+    time: string,
+};
+
+const messageCode = 'engagement-banner';
+
+const getTimestampOfLastBannerDeploy = (): Promise<string> =>
+    fetchJson('/reader-revenue/contributions-banner-deploy-log', {
+        mode: 'cors',
+    }).then((resp: BannerDeployLog) => resp && resp.time);
+
+const hasBannerBeenRedeployedSinceClosed = (): Promise<boolean> =>
+    getTimestampOfLastBannerDeploy()
+        .then(timestamp => {
+            const bannerLastDeployedAt = new Date(timestamp);
+            const userLastClosedBannerAt = userPrefs.get(
+                'engagementBannerLastClosedAt'
+            );
+
+            if (!userLastClosedBannerAt) {
```

Comment:
I had another thought about this! If we do this check before fetching the timestamp from S3, we could probably save a lot of requests to the server - first-time visitors (different browser, incognito etc) won't bother requesting the timestamp until the user has actually closed the banner in that browser.

This might also help to stagger the requests a bit when we deploy - rather than every single browser requesting straight away, they wouldn't request until the pageview after the banner's been closed

Reply:
Good idea! Sensible to reduce no. of requests as you said (even if most will be hitting the cache).